### DISCUSSION TITLE
x-html is rendering a string of html

### DISCUSSION THREAD

**Discussion by Ravenna** [alpinejs/alpine#3477] [1. Help]
I have an API I am getting my x-data from.  It all works great except for some HTML.

My HTML is coming from the API encoded. Example: `&lt;div class=&quot;content-intro&quot;&gt;&lt;p&gt;&lt;strong&gt;`

I am also nested inside two `<template x-for"">`'s.

If I use x-text is shows the same as the encoding above: `&lt;div class=&quot;content-intro&quot;&gt;&lt;p&gt;&lt;strong&gt;`

If I use x-html I get '<div class="content-intro"><p><strong>' printed on the page like in the screenshot
![Screenshot 2023-03-21 at 9 56 48 AM](https://user-images.githubusercontent.com/72239/226684729-a3825ced-feba-4e05-82f4-821f9e78ef66.png)

Basically its properly handle the HTML except its outputting it in a string. 

How can I have it output as code so that the end user isn't seeing code?

**Comment by ekwoka** [comment]
You aren't providing it HTML. You're providing it a string to add as a text node.

You need to not have the `<>` and stuff encoded to be HTML safe. It makes it not HTML.

**Comment by Ravenna** [comment]
@ekwoka thanks for the speedy response. 

I don't totally follow you but I am trying. 

The API is returning: 
![Screenshot 2023-03-21 at 11 06 56 AM](https://user-images.githubusercontent.com/72239/226701890-bdaf7cb7-f81a-4f45-8bc4-59841e2eb91b.png)

So when I then set that value on the div with x-html.  Should I be doing some other modification on that value before using it in x-html?

**Reply by ekwoka** [reply]
Well, that is an html safe string.

It's not html.

Seems like the API might be built wrong.

The simple way is to use the Dom parser API to parse the html safe string into an html string and then use that in x-html.

**Comment by Ravenna** [comment]
Got it. Thank you
